src/main/java/models/igp/ospf/OSPFPrefix.java

```diff
@@ -2,17 +2,14 @@
 
 import models.igp.IGPPrefix;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.*;
+import util.Attribute;
 
 public class OSPFPrefix extends IGPPrefix {
-    public Set<String> routerIds;
+    // routerId => attribute
+    public HashMap<String, Attribute> attributesForRouter;
 
-    public OSPFPrefix() {
-        this.routerIds = new HashSet<>();
-    }
-
-    public void addRouter(String routerId) {
```

Comment:
If we remove this function, we should replace it with something similar, to add a RouterId, Attribute to the HashMap